<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Map functions — map_ • KGExplorer</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Map functions — map_"><meta property="og:description" content="Functions to map IDs across ontologies/databases."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">KGExplorer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/KGExplorer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/KGExplorer/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Map functions</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/KGExplorer/blob/HEAD/R/0docs.R" class="external-link"><code>R/0docs.R</code></a>, <a href="https://github.com/neurogenomics/KGExplorer/blob/HEAD/R/map_colors.R" class="external-link"><code>R/map_colors.R</code></a>, <a href="https://github.com/neurogenomics/KGExplorer/blob/HEAD/R/map_genes_monarch.R" class="external-link"><code>R/map_genes_monarch.R</code></a>, and 5 more</small>
    <div class="hidden name"><code>map_.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Functions to map IDs across ontologies/databases.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">map_colors</span><span class="op">(</span></span>
<span>  <span class="va">dat</span>,</span>
<span>  columns <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  as <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"vector"</span>, <span class="st">"dict"</span>, <span class="st">"name"</span>, <span class="st">"function"</span><span class="op">)</span>,</span>
<span>  what <span class="op">=</span> <span class="st">"nodes"</span>,</span>
<span>  preferred_palettes <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">map_genes_monarch</span><span class="op">(</span></span>
<span>  <span class="va">dat</span>,</span>
<span>  <span class="va">gene_col</span>,</span>
<span>  as_graph <span class="op">=</span> <span class="fu">methods</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/methods/is.html" class="external-link">is</a></span><span class="op">(</span><span class="va">dat</span>, <span class="st">"tbl_graph"</span><span class="op">)</span>,</span>
<span>  map_by_merge <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  all.x <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">map_medgen</span><span class="op">(</span><span class="va">dat</span>, <span class="va">input_col</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">map_mondo</span><span class="op">(</span></span>
<span>  <span class="va">dat</span>,</span>
<span>  input_col <span class="op">=</span> <span class="st">"id"</span>,</span>
<span>  output_col <span class="op">=</span> <span class="st">"mondo_id"</span>,</span>
<span>  to <span class="op">=</span> <span class="st">"mondo"</span>,</span>
<span>  map_types <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  map_to <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  top_n <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  add_name <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  add_definitions <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  all.x <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  allow.cartesian <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  save_dir <span class="op">=</span> <span class="fu"><a href="utils_.html">cache_dir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">map_ontology_terms</span><span class="op">(</span></span>
<span>  <span class="va">ont</span>,</span>
<span>  terms <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  to <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"name"</span>, <span class="st">"id"</span><span class="op">)</span>,</span>
<span>  keep_order <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  invert <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">map_upheno</span><span class="op">(</span></span>
<span>  pheno_map_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"upheno"</span>, <span class="st">"monarch"</span><span class="op">)</span>,</span>
<span>  gene_map_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"monarch"</span><span class="op">)</span>,</span>
<span>  subset_db1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HP"</span><span class="op">)</span>,</span>
<span>  terms <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  fill_scores <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  show_plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  force_new <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  save_dir <span class="op">=</span> <span class="fu"><a href="utils_.html">cache_dir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">map_variants</span><span class="op">(</span></span>
<span>  <span class="va">gr</span>,</span>
<span>  build <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GRCh37"</span>, <span class="st">"GRCh38"</span><span class="op">)</span>,</span>
<span>  upstream <span class="op">=</span> <span class="fl">2000L</span>,</span>
<span>  downstream <span class="op">=</span> <span class="fl">200L</span>,</span>
<span>  keep_chr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">22</span><span class="op">)</span>, <span class="st">"X"</span>, <span class="st">"Y"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  ignore.strand <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="source">
    <h2>Source</h2>
    <p>https://data.monarchinitiative.org/upheno2/current/qc/index</p>
<p>https://data.monarchinitiative.org/upheno2/current/upheno-release/all/index.html</p>
    </div>
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>dat</dt>
<dd><p><a href="https://rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a> with genes.</p></dd>


<dt>columns</dt>
<dd><p>Names of columns to map colour palettes to.</p></dd>


<dt>as</dt>
<dd><p>A character string specifying the format to convert to.</p></dd>


<dt>what</dt>
<dd><p>What should get activated? Possible values are <code>nodes</code> or
<code>edges</code>.</p></dd>


<dt>preferred_palettes</dt>
<dd><p>Preferred palettes to use for each column.</p></dd>


<dt>gene_col</dt>
<dd><p>Name of the gene column in <code>dat</code>.</p></dd>


<dt>as_graph</dt>
<dd><p>Return the object as a <a href="https://tidygraph.data-imaginist.com/reference/tbl_graph.html" class="external-link">tbl_graph</a>.</p></dd>


<dt>map_by_merge</dt>
<dd><p>Map orthologs by merging the node data such that the 
orthologous genes will appear as a new column (<code>TRUE</code>). 
Otherwise, the orthologs will be added as new nodes to the graph 
(<code>FALSE</code>).</p></dd>


<dt>all.x</dt>
<dd><p>logical; if <code>TRUE</code>, rows from <code>x</code> which have no matching row
in <code>y</code> are included. These rows will have 'NA's in the columns that are usually
filled with values from <code>y</code>. The default is <code>FALSE</code> so that only rows with
data from both <code>x</code> and <code>y</code> are included in the output.</p></dd>


<dt>input_col</dt>
<dd><p>Column name of input IDs.</p></dd>


<dt>...</dt>
<dd><p>Arguments passed on to <code>map_mondo</code></p><dl><dt><code></code></dt>
<dd></dd>

  
</dl></dd>


<dt>output_col</dt>
<dd><p>Column name of output IDs.</p></dd>


<dt>to</dt>
<dd><p>Character vector of database(s) to map IDs to.
When not <code>"mondo"</code>, can supply multiple alternative databases to map to
 (e.g. <code>c("OMIM","Orphanet","DECIPHER")</code>).</p></dd>


<dt>map_types</dt>
<dd><p>Mapping types to include.</p></dd>


<dt>map_to</dt>
<dd><p>Mapping outputs to include
(from Mondo IDs to another database's IDs).</p></dd>


<dt>top_n</dt>
<dd><p>Top number of mappings to return per <code>top_by</code> grouping.
Set to <code>NULL</code> to skip this step.</p></dd>


<dt>add_name</dt>
<dd><p>Logical, if TRUE, add mondo name column.</p></dd>


<dt>add_definitions</dt>
<dd><p>logical, if TRUE, add mondo definition column.</p></dd>


<dt>allow.cartesian</dt>
<dd><p>See <code>allow.cartesian</code> in <code><a href="https://rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">[.data.table</a></code>.</p></dd>


<dt>save_dir</dt>
<dd><p>Directory to save cached data.</p></dd>


<dt>ont</dt>
<dd><p>An ontology of class <a href="https://rdrr.io/pkg/simona/man/ontology_DAG-class.html" class="external-link">ontology_DAG</a>.</p></dd>


<dt>terms</dt>
<dd><p>A subset of HPO IDs to include in the final dataset and plots
(e.g. c("HP:0001508","HP:0001507")).</p></dd>


<dt>keep_order</dt>
<dd><p>Return a named list of the same length and order
as <code>terms</code>.
If <code>FALSE</code>, return a named list of only the unique <code>terms</code>,
sometimes in a different order.</p></dd>


<dt>invert</dt>
<dd><p>Invert the keys/values of the dictionary,
such that the key becomes the values (and vice versa).</p></dd>


<dt>pheno_map_method</dt>
<dd><p>Method to use for mapping phenotypes across ontologies.</p><ul><li><p>"upheno"Use uPheno's phenotype-to-phenotype mappings.
 Contains fewer ontologies but with greater coverage of phenotypes.</p></li>
<li><p>"monarch"Use Monarch's phenotype-to-phenotype mappings.
Contains more ontologies but with less coverage of phenotypes.</p></li>
</ul></dd>


<dt>gene_map_method</dt>
<dd><p>Method to use for mapping genes across species.</p><ul><li><p>"monarch"Use Monarch's gene-to-gene mappings.</p></li>
</ul></dd>


<dt>subset_db1</dt>
<dd><p>Subset of ontologies to include in the plot.</p></dd>


<dt>fill_scores</dt>
<dd><p>Fill missing scores in the "equivalence_score" and
"subclass_score" columns with this value. These columns represent the
quality of mapping between two phenotypes on a scale from 0-1.</p></dd>


<dt>show_plot</dt>
<dd><p>Show the plot.</p></dd>


<dt>force_new</dt>
<dd><p>Force new data to be downloaded and processed.</p></dd>


<dt>gr</dt>
<dd><p>A <a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a> object.</p></dd>


<dt>build</dt>
<dd><p>Genome build to use when mapping genomic coordinates.</p></dd>


<dt>upstream, downstream</dt>
<dd><p>Single <code>integer</code> values representing the number of base pairs 
    upstream of the 5'-end and downstream of the 3'-end. Used in contructing 
    <code>PromoterVariants()</code> and <code>IntergenicVariants()</code> objects only.</p></dd>


<dt>keep_chr</dt>
<dd><p>Which chromosomes to keep.</p></dd>


<dt>ignore.strand</dt>
<dd><p>A <code>logical</code> indicating if strand should be
    ignored when performing overlaps.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>Mapped data.</p>


<p>Mapped <code>dat</code></p>



<p>Character vector</p>


<p>A list containing the data and plot.</p>
    </div>
    <div id="functions">
    <h2>Functions</h2>
    
<ul><li><p><code>map_colors()</code>: map_</p></li>
<li><p><code>map_genes_monarch()</code>: map_
Map Monarch genes</p>
<p>Map Monarch gene IDs to HGNC gene symbols, within or across species.</p></li>
<li><p><code>map_medgen()</code>: map_
Map Medgen.</p></li>
<li><p><code>map_mondo()</code>: map_
Map to/from mondo IDs</p></li>
<li><p><code>map_ontology_terms()</code>: map_
Map ontology terms to an alternative name.</p>
<p>Harmonise a mixed vector of term names (e.g. "Focal motor seizure")
and term IDs (e.g. c("HP:0000002","HP:0000003")).</p></li>
<li><p><code>map_upheno()</code>: map_
Map phenotypes across uPheno</p>
<p>Map phenotypes across species within the Unified Phenotype Ontology (uPheno).
First, gathers phenotype-phenotype mappings across ontologies.
Next, gathers all phenotype-gene associations for each ontology,
converts all genes to human HGNC orthologs, and computes the number of
overlapping orthologs between all pairs mapped phenotypes.
Finally, plots the results as the proportion of intersecting genes between
all pairs of phenotypes.</p></li>
<li><p><code>map_variants()</code>: map_</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu">map_colors</span><span class="op">(</span>dat<span class="op">=</span><span class="va">mtcars</span>, columns<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cyl"</span>,<span class="st">"gear"</span><span class="op">)</span>, preferred<span class="op">=</span><span class="st">"viridis"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using palette: viridis</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using palette: okabe</span>
<span class="r-in"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="utils_.html">example_dat</a></span><span class="op">(</span><span class="st">"gene"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dt2</span> <span class="op">&lt;-</span> <span class="fu">map_genes_monarch</span><span class="op">(</span>dat<span class="op">=</span><span class="va">dat</span>, gene_col<span class="op">=</span><span class="st">"gene"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering with `queries`.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Files found: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing 1 Monarch files.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> - 1/1: gene_homology.all</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Unique species with orthologs: 25</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtered 'subject_db' : 6,503,843 / 6,906,371 rows dropped.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Unique orthologs: 314,817</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 6 / 6 rows remain after gene orthology mapping.</span>
<span class="r-in"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="utils_.html">example_dat</a></span><span class="op">(</span>rm_types<span class="op">=</span><span class="st">"gene"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dat2</span> <span class="op">&lt;-</span> <span class="fu">map_mondo</span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat</span>, map_to<span class="op">=</span><span class="st">"hpo"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required namespace: echogithub</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preexisting file detected. Set force_overwrite=TRUE to override this.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preexisting file detected. Set force_overwrite=TRUE to override this.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preexisting file detected. Set force_overwrite=TRUE to override this.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preexisting file detected. Set force_overwrite=TRUE to override this.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preexisting file detected. Set force_overwrite=TRUE to override this.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preexisting file detected. Set force_overwrite=TRUE to override this.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preexisting file detected. Set force_overwrite=TRUE to override this.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preexisting file detected. Set force_overwrite=TRUE to override this.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preexisting file detected. Set force_overwrite=TRUE to override this.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preexisting file detected. Set force_overwrite=TRUE to override this.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preexisting file detected. Set force_overwrite=TRUE to override this.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preexisting file detected. Set force_overwrite=TRUE to override this.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mapping id --&gt; mondo_id</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> All local files already up-to-date!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing cached file: /github/home/.cache/R/KGExplorer/mondo.owl</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding term metadata.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> IC_method: IC_offspring</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ancestor metadata.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting absolute ontology level for 31,550 IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2412 ancestors found at level 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Translating all terms to names.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Returning a vector of terms (same order as input).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converted ontology to: adjacency </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting absolute ontology level for 31,550 IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4 / 21 (19.05%) mondo_id missing.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 4 / 21 (19.05%) mondo_name missing.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 14 / 21 (66.67%) mondo_def missing.</span>
<span class="r-in"><span><span class="va">ont</span> <span class="op">&lt;-</span> <span class="fu"><a href="get_.html">get_ontology</a></span><span class="op">(</span><span class="st">"hp"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> All local files already up-to-date!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Importing cached file: /github/home/.cache/R/KGExplorer/hp-international.owl</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding term metadata.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> IC_method: IC_offspring</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding ancestor metadata.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting absolute ontology level for 25,301 IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 900 ancestors found at level 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Translating all terms to names.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Returning a vector of terms (same order as input).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converted ontology to: adjacency </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Getting absolute ontology level for 25,301 IDs.</span>
<span class="r-in"><span><span class="va">terms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Focal motor seizure"</span>,<span class="st">"HP:0000002"</span>,<span class="st">"HP:0000003"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">term_names</span> <span class="op">&lt;-</span> <span class="fu">map_ontology_terms</span><span class="op">(</span>ont<span class="op">=</span><span class="va">ont</span>, terms<span class="op">=</span><span class="va">terms</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Translating all terms to names.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Returning a vector of terms (same order as input).</span>
<span class="r-in"><span><span class="va">term_ids</span> <span class="op">&lt;-</span> <span class="fu">map_ontology_terms</span><span class="op">(</span>ont<span class="op">=</span><span class="va">ont</span>, terms<span class="op">=</span><span class="va">terms</span>, to<span class="op">=</span><span class="st">"id"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Translating all terms to HPO IDs.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> + Returning a vector of terms (same order as input).</span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">map_upheno</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span></span>
<span class="r-in"><span><span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span><span class="st">"1:100-10000"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu">map_variants</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

