<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Map uPheno data — map_upheno_data • KGExplorer</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Map uPheno data — map_upheno_data"><meta property="og:description" content="Get uPheno cross-species mapping data by:
Downloading cross-species phenotype-phenotype mappings.
Downloading within-spceies phenotype-gene mappings,
and converting these genes to human orthologs.
Merging the phenotype-phenotype and phenotype-gene mappings.
Filtering out any mappings that do not have a human ortholog within
each respective phenotype.
Calculating the proportion of orthologous genes overlapping across
the species-specific phenotype-gene mappings.
Iterating the above steps using multiple methods
(pheno_map_method) and concatenating the results together.

"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">KGExplorer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.05</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/KGExplorer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/neurogenomics/KGExplorer/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Map uPheno data</h1>
    <small class="dont-index">Source: <a href="https://github.com/neurogenomics/KGExplorer/blob/improve-docs/R/map_upheno_data.R" class="external-link"><code>R/map_upheno_data.R</code></a></small>
    <div class="hidden name"><code>map_upheno_data.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Get uPheno cross-species mapping data by:</p><ul><li><p>Downloading cross-species phenotype-phenotype mappings.</p></li>
<li><p>Downloading within-spceies phenotype-gene mappings,
and converting these genes to human orthologs.</p></li>
<li><p>Merging the phenotype-phenotype and phenotype-gene mappings.</p></li>
<li><p>Filtering out any mappings that do not have a human ortholog within
each respective phenotype.</p></li>
<li><p>Calculating the proportion of orthologous genes overlapping across
the species-specific phenotype-gene mappings.</p></li>
<li><p>Iterating the above steps using multiple methods
(<code>pheno_map_method</code>) and concatenating the results together.</p></li>
</ul></div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">map_upheno_data</span><span class="op">(</span></span>
<span>  pheno_map_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"upheno"</span>, <span class="st">"monarch"</span><span class="op">)</span>,</span>
<span>  gene_map_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"monarch"</span><span class="op">)</span>,</span>
<span>  keep_nogenes <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  fill_scores <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  terms <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  save_dir <span class="op">=</span> <span class="fu"><a href="cache_.html">cache_dir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  agg <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  force_new <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-pheno-map-method">pheno_map_method<a class="anchor" aria-label="anchor" href="#arg-pheno-map-method"></a></dt>
<dd><p>Method to use for mapping phenotypes across ontologies.</p><ul><li><p>"upheno"Use uPheno's phenotype-to-phenotype mappings.
 Contains fewer ontologies but with greater coverage of phenotypes.</p></li>
<li><p>"monarch"Use Monarch's phenotype-to-phenotype mappings.
Contains more ontologies but with less coverage of phenotypes.</p></li>
</ul></dd>


<dt id="arg-gene-map-method">gene_map_method<a class="anchor" aria-label="anchor" href="#arg-gene-map-method"></a></dt>
<dd><p>Method to use for mapping genes across species.</p><ul><li><p>"monarch"Use Monarch's gene-to-gene mappings.</p></li>
</ul></dd>


<dt id="arg-keep-nogenes">keep_nogenes<a class="anchor" aria-label="anchor" href="#arg-keep-nogenes"></a></dt>
<dd><p>Logical indicating whether to keep mappings that do not
have any orthologous genes.</p></dd>


<dt id="arg-fill-scores">fill_scores<a class="anchor" aria-label="anchor" href="#arg-fill-scores"></a></dt>
<dd><p>Fill missing scores in the "equivalence_score" and
"subclass_score" columns with this value. These columns represent the
quality of mapping between two phenotypes on a scale from 0-1.</p></dd>


<dt id="arg-terms">terms<a class="anchor" aria-label="anchor" href="#arg-terms"></a></dt>
<dd><p>A subset of HPO IDs to include in the final dataset and plots
(e.g. c("HP:0001508","HP:0001507")).</p></dd>


<dt id="arg-save-dir">save_dir<a class="anchor" aria-label="anchor" href="#arg-save-dir"></a></dt>
<dd><p>Directory to save cached data.</p></dd>


<dt id="arg-agg">agg<a class="anchor" aria-label="anchor" href="#arg-agg"></a></dt>
<dd><p>Aggregate the data to to phenotype-level (TRUE).
Otherwise, keep data at gene level (FALSE).</p></dd>


<dt id="arg-force-new">force_new<a class="anchor" aria-label="anchor" href="#arg-force-new"></a></dt>
<dd><p>Create a new file instead of using any cached files.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A data.table containing the mapped data.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="va">pheno_map_genes_match</span> <span class="op">&lt;-</span> <span class="fu">map_upheno_data</span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

