<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Get started â€¢ KGExplorer</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Get started">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">KGExplorer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.09</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/KGExplorer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/docker.html">Docker/Singularity Containers</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/neurogenomics/KGExplorer/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Get started</h1>
                        <h4 data-toc-skip class="author"><h4>
<br>
Authors: <i>Brian Schilder, Hiranyamaya Dash</i><br>
</h4></h4>
            
            <h4 data-toc-skip class="date"><h4>
<br>
Vignette updated: <i>Aug-02-2025</i><br>
</h4></h4>
      
      <small class="dont-index">Source: <a href="https://github.com/neurogenomics/KGExplorer/blob/master/vignettes/KGExplorer.Rmd" class="external-link"><code>vignettes/KGExplorer.Rmd</code></a></small>
      <div class="hidden name"><code>KGExplorer.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>KGExplorer is an R package designed for efficient exploration and
manipulation of biomedical knowledge graphs and ontologies. Its core
functionalities include:</p>
<ul>
<li><p><strong>Graph Manipulation and Visualization</strong>: Tools like
<code><a href="../reference/plot_.html">plot_graph_3d()</a></code> and <code><a href="../reference/plot_.html">plot_ggnetwork()</a></code> enable
intuitive visualization and analysis of ontological data.</p></li>
<li><p><strong>Data Retrieval</strong>: Functions such as
<code><a href="../reference/get_.html">get_mondo_maps()</a></code> and <code><a href="../reference/get_.html">get_ontology()</a></code> facilitate
fetching data from prominent biomedical databases.</p></li>
<li><p><strong>ID Mapping</strong>: Mapping utilities like
<code><a href="../reference/map_.html">map_genes_monarch()</a></code> and <code><a href="../reference/map_.html">map_mondo()</a></code> allow
seamless conversion between various identifier systems.</p></li>
<li><p><strong>Graph Filtering</strong>: With functions like
<code><a href="../reference/filter_.html">filter_graph()</a></code> and <code><a href="../reference/filter_.html">filter_kg()</a></code>, users can
subset knowledge graphs to focus on relevant subsets.</p></li>
<li><p><strong>Graph Conversion</strong>: Utilities such as
<code><a href="../reference/to_.html">graph_to_dt()</a></code> and <code><a href="../reference/graph_to_plotly.html">graph_to_plotly()</a></code> provide
flexibility to transform knowledge graphs into different data formats or
visualization-ready structures.</p></li>
</ul>
<p>The package leverages robust R libraries such as
<code>tidygraph</code>, <code>data.table</code>, and popular plotting
tools including <code>plotly</code> and <code>ggplot2</code>. KGExplorer
adopts a modular design, with dedicated functions for data retrieval,
manipulation, visualization, and utility operations. Comprehensive
caching support ensures efficient handling of large datasets, reducing
the need for repeated downloads.</p>
</div>
<div class="section level2">
<h2 id="use-cases">Use cases<a class="anchor" aria-label="anchor" href="#use-cases"></a>
</h2>
<p>The following examples illustrate how KGExplorer can be used to
extract meaningful insights from biomedical knowledge graphs. Each use
case highlights key functionalities of the package and demonstrates
practical applications in biomedical research.</p>
<p>To explore additional functionalities and find detailed documentation
for all available functions, users can refer to the <a href="https://neurogenomics.github.io/KGExplorer/reference/index.html" class="external-link">KGExplorer
reference guide</a>.</p>
<div class="section level3">
<h3 id="extract-diseasephenotype-cell-type-associations">Extract disease/phenotype-cell type associations<a class="anchor" aria-label="anchor" href="#extract-diseasephenotype-cell-type-associations"></a>
</h3>
<p>The package provides functionality to extract associations between
diseases/phenotypes and cell types. This can help researchers understand
the relationships and interactions between different biological
entities.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get the Monarch knowledge graph</span></span>
<span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_.html">get_monarch_kg</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter the graph to include only edges between diseases/phenotypes and cell</span></span>
<span><span class="co"># types</span></span>
<span><span class="va">g2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_.html">filter_kg</a></span><span class="op">(</span><span class="va">g</span>, </span>
<span>                to_categories <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"biolink:Disease"</span>,</span>
<span>                                  <span class="st">"biolink:PhenotypicFeature"</span><span class="op">)</span>,</span>
<span>                from_categories <span class="op">=</span> <span class="st">"biolink:Cell"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot the filtered graph using visNetwork</span></span>
<span><span class="fu"><a href="../reference/plot_graph_visnetwork.html">plot_graph_visnetwork</a></span><span class="op">(</span><span class="va">g2</span>,</span>
<span>                      selectedBy <span class="op">=</span> <span class="st">"id"</span>,</span>
<span>                      label_var <span class="op">=</span> <span class="st">"name"</span>,</span>
<span>                      layout <span class="op">=</span> <span class="st">"layout_nicely"</span>,</span>
<span>                      colour_var <span class="op">=</span> <span class="st">"category"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="assess-known-animal-models-of-human-phenotypes">Assess known animal models of human phenotypes<a class="anchor" aria-label="anchor" href="#assess-known-animal-models-of-human-phenotypes"></a>
</h3>
<p>The package includes tools to assess known animal models of human
phenotypes, which can be crucial for translational research and
understanding disease mechanisms.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Map uPheno data</span></span>
<span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/map_upheno_data.html">map_upheno_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot the mapped uPheno data</span></span>
<span><span class="va">upheno_plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_upheno.html">plot_upheno</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="link-diseases-to-phenotypes-to-genes-to-variants">Link diseases to phenotypes to genes to variants<a class="anchor" aria-label="anchor" href="#link-diseases-to-phenotypes-to-genes-to-variants"></a>
</h3>
<p>Users can link diseases to phenotypes, genes, and variants, providing
a comprehensive view of the genetic and phenotypic landscape. This can
aid in identifying potential genetic markers and understanding the
genetic basis of diseases.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Link Monarch data to create a graph with variant-disease, variant-phenotype,</span></span>
<span><span class="co"># and variant-gene associations</span></span>
<span><span class="va">gm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/link_.html">link_monarch</a></span><span class="op">(</span>maps <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"variant"</span>, <span class="st">"disease"</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"variant"</span>, <span class="st">"phenotype"</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"variant"</span>, <span class="st">"gene"</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Join the linked graph with the original graph</span></span>
<span><span class="va">gm2</span> <span class="op">&lt;-</span> <span class="fu">tidygraph</span><span class="fu">::</span><span class="fu"><a href="https://tidygraph.data-imaginist.com/reference/graph_join.html" class="external-link">graph_join</a></span><span class="op">(</span><span class="va">gm</span>,<span class="va">g</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter the graph to include only specific categories of nodes</span></span>
<span><span class="va">gm2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_.html">filter_graph</a></span><span class="op">(</span><span class="va">gm2</span>,</span>
<span>                    node_filters <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>category<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"disease"</span>,</span>
<span>                                                   <span class="st">"phenotype"</span>,</span>
<span>                                                   <span class="st">"phenotypicfeature"</span>,</span>
<span>                                                   <span class="st">"gene"</span>,</span>
<span>                                                   <span class="st">"variant"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Further filter the graph to limit the size</span></span>
<span><span class="va">gm3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_.html">filter_graph</a></span><span class="op">(</span>g <span class="op">=</span> <span class="va">gm2</span>, size <span class="op">=</span> <span class="fl">20000</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Update vertex attributes for category and name</span></span>
<span><span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/vertex_attr.html" class="external-link">vertex_attr</a></span><span class="op">(</span><span class="va">gm3</span>,<span class="st">"category"</span><span class="op">)</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/chartr.html" class="external-link">tolower</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"biolink:"</span>,<span class="st">""</span>, <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/vertex_attr.html" class="external-link">vertex_attr</a></span><span class="op">(</span><span class="va">gm3</span>,<span class="st">"category"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/vertex_attr.html" class="external-link">vertex_attr</a></span><span class="op">(</span><span class="va">gm3</span>,<span class="st">"name"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/vertex_attr.html" class="external-link">vertex_attr</a></span><span class="op">(</span><span class="va">gm3</span>,<span class="st">"id"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot the final graph using visNetwork</span></span>
<span><span class="fu"><a href="../reference/plot_graph_visnetwork.html">plot_graph_visnetwork</a></span><span class="op">(</span><span class="va">gm3</span>, </span>
<span>                      selectedBy <span class="op">=</span> <span class="st">"category"</span>,</span>
<span>                      label_var <span class="op">=</span> <span class="st">"name"</span>,</span>
<span>                      layout <span class="op">=</span> <span class="st">"layout_nicely"</span>,</span>
<span>                      colour_var <span class="op">=</span> <span class="st">"category"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<details><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] KGExplorer_0.99.09</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] grr_0.9.5                 httr2_1.2.1              </span></span>
<span><span class="co">##   [3] rlang_1.1.6               magrittr_2.0.3           </span></span>
<span><span class="co">##   [5] clue_0.3-66               GetoptLong_1.0.5         </span></span>
<span><span class="co">##   [7] matrixStats_1.5.0         compiler_4.5.1           </span></span>
<span><span class="co">##   [9] png_0.1-8                 systemfonts_1.2.3        </span></span>
<span><span class="co">##  [11] vctrs_0.6.5               maps_3.4.3               </span></span>
<span><span class="co">##  [13] gprofiler2_0.2.3          stringr_1.5.1            </span></span>
<span><span class="co">##  [15] rvest_1.0.4               shape_1.4.6.1            </span></span>
<span><span class="co">##  [17] pkgconfig_2.0.3           crayon_1.5.3             </span></span>
<span><span class="co">##  [19] fastmap_1.2.0             backports_1.5.0          </span></span>
<span><span class="co">##  [21] promises_1.3.3            rmarkdown_2.29           </span></span>
<span><span class="co">##  [23] ragg_1.4.0                purrr_1.1.0              </span></span>
<span><span class="co">##  [25] xfun_0.52                 cachem_1.1.0             </span></span>
<span><span class="co">##  [27] pals_1.10                 aplot_0.2.8              </span></span>
<span><span class="co">##  [29] jsonlite_2.0.0            later_1.4.2              </span></span>
<span><span class="co">##  [31] BiocParallel_1.43.4       cluster_2.1.8.1          </span></span>
<span><span class="co">##  [33] broom_1.0.9               parallel_4.5.1           </span></span>
<span><span class="co">##  [35] R6_2.6.1                  rols_3.5.0               </span></span>
<span><span class="co">##  [37] stringi_1.8.7             bslib_0.9.0              </span></span>
<span><span class="co">##  [39] RColorBrewer_1.1-3        car_3.1-3                </span></span>
<span><span class="co">##  [41] jquerylib_0.1.4           Rcpp_1.1.0               </span></span>
<span><span class="co">##  [43] iterators_1.0.14          knitr_1.50               </span></span>
<span><span class="co">##  [45] IRanges_2.43.0            httpuv_1.6.16            </span></span>
<span><span class="co">##  [47] igraph_2.1.4              Matrix_1.7-3             </span></span>
<span><span class="co">##  [49] tidyselect_1.2.1          dichromat_2.0-0.1        </span></span>
<span><span class="co">##  [51] abind_1.4-8               yaml_2.3.10              </span></span>
<span><span class="co">##  [53] doParallel_1.0.17         codetools_0.2-20         </span></span>
<span><span class="co">##  [55] lattice_0.22-7            tibble_3.3.0             </span></span>
<span><span class="co">##  [57] shiny_1.11.1              Biobase_2.69.0           </span></span>
<span><span class="co">##  [59] treeio_1.33.0             evaluate_1.0.4           </span></span>
<span><span class="co">##  [61] gridGraphics_0.5-1        desc_1.4.3               </span></span>
<span><span class="co">##  [63] xml2_1.3.8                circlize_0.4.16          </span></span>
<span><span class="co">##  [65] pillar_1.11.0             ggtree_3.17.1            </span></span>
<span><span class="co">##  [67] ggpubr_0.6.1              carData_3.0-5            </span></span>
<span><span class="co">##  [69] foreach_1.5.2             stats4_4.5.1             </span></span>
<span><span class="co">##  [71] ggfun_0.2.0               plotly_4.11.0            </span></span>
<span><span class="co">##  [73] generics_0.1.4            S4Vectors_0.47.0         </span></span>
<span><span class="co">##  [75] ggplot2_3.5.2             scales_1.4.0             </span></span>
<span><span class="co">##  [77] tidytree_0.4.6            xtable_1.8-4             </span></span>
<span><span class="co">##  [79] glue_1.8.0                orthogene_1.15.0         </span></span>
<span><span class="co">##  [81] mapproj_1.2.12            scatterplot3d_0.3-44     </span></span>
<span><span class="co">##  [83] lazyeval_0.2.2            tools_4.5.1              </span></span>
<span><span class="co">##  [85] data.table_1.17.8         ggsignif_0.6.4           </span></span>
<span><span class="co">##  [87] babelgene_22.9            fs_1.6.6                 </span></span>
<span><span class="co">##  [89] tidygraph_1.3.1           fastmatch_1.1-6          </span></span>
<span><span class="co">##  [91] grid_4.5.1                tidyr_1.3.1              </span></span>
<span><span class="co">##  [93] ape_5.8-1                 colorspace_2.1-1         </span></span>
<span><span class="co">##  [95] nlme_3.1-168              patchwork_1.3.1          </span></span>
<span><span class="co">##  [97] homologene_1.4.68.19.3.27 Formula_1.2-5            </span></span>
<span><span class="co">##  [99] cli_3.6.5                 rappdirs_0.3.3           </span></span>
<span><span class="co">## [101] Polychrome_1.5.4          textshaping_1.0.1        </span></span>
<span><span class="co">## [103] viridisLite_0.4.2         ComplexHeatmap_2.25.2    </span></span>
<span><span class="co">## [105] dplyr_1.1.4               gtable_0.3.6             </span></span>
<span><span class="co">## [107] rstatix_0.7.2             yulab.utils_0.2.0        </span></span>
<span><span class="co">## [109] sass_0.4.10               digest_0.6.37            </span></span>
<span><span class="co">## [111] BiocGenerics_0.55.1       ggplotify_0.1.2          </span></span>
<span><span class="co">## [113] rjson_0.2.23              htmlwidgets_1.6.4        </span></span>
<span><span class="co">## [115] farver_2.1.2              htmltools_0.5.8.1        </span></span>
<span><span class="co">## [117] pkgdown_2.1.3             simona_1.7.1             </span></span>
<span><span class="co">## [119] lifecycle_1.0.4           httr_1.4.7               </span></span>
<span><span class="co">## [121] mime_0.13                 GlobalOptions_0.1.2</span></span></code></pre>
</details><p><br></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Brian Schilder.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer>
</div>






  </body>
</html>
